src/Elrond/Codec.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Codec.h"
+
+#include "HexCoding.h"
+
+using namespace TW;
+using namespace TW::Elrond;
+
+std::string Codec::encodeString(const std::string& value) {
+    std::string encoded = hex(TW::data(value));
+    return encoded;
+}
+
+std::string Codec::encodeUint64(uint64_t value) {
+    std::string encoded = normalizeHex(hex(value));
+    return encoded;
+}
+
+std::string Codec::encodeBigInt(const std::string& value) {
+    return encodeBigInt(uint256_t(value));
+}
+
+// For reference, see https://docs.elrond.com/developers/developer-reference/elrond-serialization-format/#arbitrary-width-big-numbers.
+std::string Codec::encodeBigInt(uint256_t value) {
+    // First, load the 256-bit value into a "Data" object.
+    auto valueAsData = Data();
+    encode256BE(valueAsData, value, 256);
+    
+    std::string encoded = normalizeHex(hex(valueAsData));
```

Comment:
I suggest using `hex(store(value, 0))`, that will have no leading 0s

Reply:
Applied the suggestion - I've called `store()` without the second & optional parameter (the second parameter is newly introduced in `master` and would only be available after _rebase_ - I've skipped the rebase, which could possibly break the _diff_ of the PR on github's code review interface).

Reply:
Rigth, the parameter was just added, 0 is the default which is the same as the previous behaviour (no padding), so it should not cause issue in merge.